sdk/tables/azure-data-tables/tests/test_table_client_cosmos.py

```diff
@@ -175,6 +183,318 @@ def test_table_name_errors_bad_length(self, tables_cosmos_account_name, tables_p
+    ):
+        base_url = self.account_url(tables_cosmos_account_name, "cosmos")
+        table_name = self.get_resource_name("mytable")
+        sas_token = self.generate_sas(
+            generate_account_sas,
+            tables_primary_cosmos_account_key,
+            resource_types=ResourceTypes.from_string("sco"),
+            permission=AccountSasPermissions.from_string("rwdlacu"),
+            expiry=datetime.utcnow() + timedelta(hours=1),
+        )
+        name_filter = "TableName eq '{}'".format(table_name)
+        conn_str = f"AccountName={tables_cosmos_account_name};AccountKey={tables_primary_cosmos_account_key.named_key.key};EndpointSuffix=cosmos.azure.com"
+
+        with TableServiceClient.from_connection_string(conn_str) as client:
+            client.create_table(table_name)
+            result = client.query_tables(name_filter)
+            assert len(list(result)) == 1
+
+        with TableServiceClient(f"{base_url}/?{sas_token}", credential=tables_primary_cosmos_account_key) as client:
+            entities = client.get_table_client(table_name).query_entities(
+                query_filter="PartitionKey eq @pk",
+                parameters={"pk": "dummy-pk"},
+            )
+            for e in entities:
+                pass
+            client.delete_table(table_name)
+
+    @cosmos_decorator
+    @recorded_by_proxy
+    def test_table_client_with_sas_token_credential(
+        self, tables_cosmos_account_name, tables_primary_cosmos_account_key
+    ):
+        base_url = self.account_url(tables_cosmos_account_name, "cosmos")
+        table_name = self.get_resource_name("mytable")
+        sas_token = self.generate_sas(
+            generate_account_sas,
+            tables_primary_cosmos_account_key,
+            resource_types=ResourceTypes.from_string("sco"),
+            permission=AccountSasPermissions.from_string("rwdlacu"),
+            expiry=datetime.utcnow() + timedelta(hours=1),
+        )
+
+        with TableClient(base_url, table_name, credential=AzureSasCredential(sas_token)) as client:
+            table = client.create_table()
+            assert table.name == table_name
+
+        with TableClient.from_table_url(f"{base_url}/{table_name}", credential=AzureSasCredential(sas_token)) as client:
+            entities = client.query_entities(
+                query_filter="PartitionKey eq @pk",
+                parameters={"pk": "dummy-pk"},
+            )
+            for e in entities:
+                pass
+
+        conn_str = (
+            f"AccountName={tables_cosmos_account_name};SharedAccessSignature={AzureSasCredential(sas_token).signature}"
+        )
+        with TableClient.from_connection_string(conn_str, table_name) as client:
+            entities = client.query_entities(
+                query_filter="PartitionKey eq @pk",
+                parameters={"pk": "dummy-pk"},
+            )
+            with pytest.raises(ClientAuthenticationError) as ex:
```

Comment:
Not sure why this one is raising any error? Might need to investigate here.

Reply:
It is the original error we get from service. We don't reprocess error in query_entities() operation.

Reply:
I think this is the only one left to investigate - not sure why the same SAS token is successful when passed in as a `SASTokenCredential` object, but not when passed in here.